pyeapi/api/interfaces.py

```diff
@@ -407,8 +411,55 @@ def delete(self, name):
             NotImplementedError: Deleting  Ethernet interfaces is not supported
 
         """
+        ver = self.node.version_number.split('.')
+        if int(ver[0]) >= 4 and int(ver[1]) >= 15:
+            return self.configure('no interface %s' % name)
         raise NotImplementedError('deleting Ethernet interfaces is '
-                                  'not supported')
+                                  'not supported in EOS version %s'
+                                  % self.node.version)
+
+    def create_subinterface(self, phys_int_name, logical_int, vlan=None):
```

Comment:
WRT the argument name, elsewhere in pyeapi, the argument `name` is used heavily when referring to the main element.  So here, the args could be (self, name, sub_intf, vid).  You'll notice `vid` is used in the vlan api to signify vlan id: https://github.com/arista-eosplus/pyeapi/blob/develop/pyeapi/api/vlans.py